blocks/small-manual-promo-block/features/small-manual-promo/default.jsx

```diff
@@ -1,105 +1,53 @@
 
-  const renderWithLink = useCallback((element, props, attributes) => (
-    <a
-      href={customFields.linkURL || '#'}
-      className={(props && props.className) || ''}
-      target={customFields.newTab ? '_blank' : '_self'}
-      rel={customFields.newTab ? 'noreferrer noopener' : ''}
-      onClick={!customFields.linkURL ? (evt) => {
-        evt.preventDefault();
-      } : undefined}
-      {...attributes}
-    >
-      {element}
-    </a>
-  ), [customFields.linkURL, customFields.newTab]);
+  const imagePosition = customFields?.imagePosition || 'right';
+  const headlineMarginClass = SmallPromoStyles(imagePosition, 'headlineMargin');
 
-  const headline = customFields.showHeadline && customFields.headline
+  const headlineOutput = customFields.showHeadline && customFields.headline
     && (
-      <div className={`promo-headline ${headlineMarginClass}`}>
-        { renderWithLink((
-          <HeadlineText
-            primaryFont={getThemeStyle(arcSite)['primary-font-family']}
-            className="sm-promo-headline"
-          >
-            {customFields.headline}
-          </HeadlineText>
-        ), { className: 'sm-promo-headline' }) }
-      </div>
+      <PromoHeadline
+        text={customFields.headline}
+        link={customFields.linkURL}
+        className={headlineMarginClass}
+        linkClassName="sm-promo-headline"
+        headingClassName="sm-promo-headline"
+        newTab={customFields.newTab}
+      />
     );
 
-  const image = showImage && customFields.imageURL && resizedImageOptions
+  const image = customFields?.showImage && customFields.imageURL
     && (
-      <div
-        className={imageMarginClass}
-        {...searchableField('imageURL')}
-      >
-        { renderWithLink(
-          <Image
-            url={customFields.imageURL}
+      <div style={{ position: isAdmin ? 'relative' : null }}>
+        <div {...searchableField('imageURL')}>
+          <PromoImage
+            customImageURL={customFields.imageURL}
             alt={customFields.headline}
-            // small should be 3:2 aspect ratio
-            {...ratios}
-            breakpoints={getProperties(arcSite)?.breakpoints}
-            resizerURL={getProperties(arcSite)?.resizerURL}
-            resizedImageOptions={resizedImageOptions}
-          />, {}, { 'aria-hidden': 'true', tabIndex: '-1' },
-        )}
+            promoSize="SM"
+            imageRatio={customFields.imageRatio}
+            linkURL={customFields.linkURL}
+            newTab={customFields.newTab}
+            lazyLoad={customFields.lazyLoad}
+          />
+        </div>
       </div>
     );
 
-  // base case for rendering image without even a link
-  return (
-    <>
-      <article className="container-fluid small-promo">
-        {getPromoContainer(headline, image, promoContainersStyles, imagePosition, isAdmin)}
-      </article>
-      <hr />
-    </>
-  );
+  return SmallPromoContainer(headlineOutput, image, imagePosition);
```

Comment:
SmallPromoContainer is a react component, right? if it's TitleCase, can we make it `<SmallPromoContainer headlineOutput={headlineOutput} ... />`? either that or make it camelCase as a helper function.

Reply:
kind of surprised the linters didn't flag this